spec/lib/site_settings/deprecated_settings_spec.rb

```diff
@@ -0,0 +1,223 @@
+# frozen_string_literal: true
+
+RSpec.describe SiteSettings::DeprecatedSettings do
+  def deprecate_override!(settings, tl_group_overrides = [])
+    @original_settings = SiteSettings::DeprecatedSettings::SETTINGS.dup
+    SiteSettings::DeprecatedSettings::SETTINGS.clear
+    SiteSettings::DeprecatedSettings::SETTINGS.push(settings)
+
+    if tl_group_overrides.any?
+      @original_override_tl_group = SiteSettings::DeprecatedSettings::OVERRIDE_TL_GROUP_SETTINGS.dup
+      SiteSettings::DeprecatedSettings::OVERRIDE_TL_GROUP_SETTINGS.clear
+      SiteSettings::DeprecatedSettings::OVERRIDE_TL_GROUP_SETTINGS.push(*tl_group_overrides)
+    end
+
+    SiteSetting.setup_deprecated_methods
+  end
+
+  after do
+    if defined?(@original_settings)
+      SiteSettings::DeprecatedSettings::SETTINGS.clear
+      SiteSettings::DeprecatedSettings::SETTINGS.concat(@original_settings)
+    end
+
+    if defined?(@original_override_tl_group)
+      SiteSettings::DeprecatedSettings::OVERRIDE_TL_GROUP_SETTINGS.clear
+      SiteSettings::DeprecatedSettings::OVERRIDE_TL_GROUP_SETTINGS.concat(
+        @original_override_tl_group,
+      )
+    end
+
+    # Necessary because Discourse.deprecate uses redis to see if the warning
+    # was already logged.
+    Discourse.redis.flushdb
```

Comment:
I wasn't sure whether I should add this to after: :each in rails helper; we do for system specs but it's unclear what the impact will be on other specs

Reply:
You can use `use_redis_snapshotting` in the example group instead of flushing manually here.

Reply:
Ahhh that's right, forgot about that, thanks!